dbms/src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -788,11 +876,26 @@ void InterpreterSelectQuery::executeFetchColumns(
 
     /// Actions to calculate ALIAS if required.
     ExpressionActionsPtr alias_actions;
-    /// Are ALIAS columns required for query execution?
-    auto alias_columns_required = false;
 
     if (storage)
     {
+        /// Append columns from the table filter to required
+        if (context.hasUserProperty(storage->getDatabaseName(), storage->getTableName(), "filter"))
+        {
+            auto initial_required_columns = required_columns;
+            ExpressionActionsPtr actions;
+            generateFilterActions(actions, storage, context, initial_required_columns);
```

Comment:
Is this the third time we calculate the same actions?

Reply:
In this call we may potentially have the different set of the `required_columns`, that's why we need a distinct invocation.